clang/test/CodeGenSYCL/code_location.cpp

```diff
@@ -0,0 +1,101 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem -sycl-std=2020 -fsycl-int-header=%t.h %s -o %t.out
+// RUN: FileCheck -input-file=%t.h %s
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem -sycl-std=2020 -DNDEBUG -fsycl-int-header=%t2.h %s -o %t2.out
+// RUN: FileCheck -input-file=%t2.h %s
+
+#include "Inputs/sycl.hpp"
+#ifndef NDEBUG
+int test1() {
+  cl::sycl::queue q;
+  q.submit([&](cl::sycl::handler &h) { h.single_task([] {}); });
+  q.submit([&](cl::sycl::handler &h) { h.single_task<class KernelName>([]() {}); });
+  return 0;
+}
+// Specializations of KernelInfo for kernel function types:
+// CHECK: template <> struct KernelInfoData<'_', 'Z', 'T', 'S', 'Z', 'Z', '5', 't', 'e', 's', 't', '1', 'v', 'E', 'N', 'K', 'U', 'l', 'R', 'N', '2', 'c', 'l', '4', 's', 'y', 'c', 'l', '7', 'h', 'a', 'n', 'd', 'l', 'e', 'r', 'E', 'E', '_', 'c', 'l', 'E', 'S', '2', '_', 'E', 'U', 'l', 'v', 'E', '_'> {
+// CHECK: static constexpr const char* getFileName() { return code_location.cpp; }
```

Comment:
This is missing the double-quotes around the filename.
```suggestion
// CHECK: static constexpr const char* getFileName() { return "code_location.cpp"; }
```